### ISSUE TITLE
[BUG] Interesting Duplicate Subscription Issue

### ISSUE THREAD

**Issue by Atreidae** [Tzahi12345/YoutubeDL-Material#801]
Found an interesting way to trigger a race condition when YTDL-M is checking for new downloads.

This causes Archive.txt and Videos.txt to flood out of control as well as saturating the downloads queue with hundreds of "Checking info" downloads that eventually causes YTDL-M to restart from a memory issue. 

**To Reproduce**
If you manage to subscribe to the same channel twice using two different URLs, For example one using the channel GUID and one using the Username, both entries will appear in the subscriptions page (As this is not sorted, its not obvious you have added a duplicate)

However both "Channels" will update to use the same "Name" string, and thus the same Video.txt and Archives.txt files.
Thus, I assume both are trying to update the channel DB with entries over and over again

**Expected behavior**
YTDL-M notices two entries with the same channel name and prompts the user warning of a possible duplicate, provide another name (in the case of actual name duplicates across channels) or to ignore and continue (when restoring subscriptions over old files)

**Workaround**
If you think you might be affected by this, a faster way than searching all your subscriptions on the subscriptions page.

Connect to your MongoDB using something like Mongo Compass and open the ytdl_material.subscriptions db

Filter your results using the "{"name":1}" filter to sort by the "Name" string.
Look for duplicate names.

When you find a duplicate, check that its the same channel using the info icon on the subscriptions page (remember, different URL's may point to the same channel)

Click unsubscribe on one of them.

**Comment by Atreidae** [comment]
(Example of duplicate's that arent detected)
https://www.youtube.com/user/XANIYO777

https://www.youtube.com/channel/UCeJ_13QGu1nelAC4ZXEtWMw

**Comment by Tzahi12345** [comment]
Good point, we just check if the URL is the same but we need to check if the name would be the same. I think I'll just append the sub ID to the name if we find out that the name is the same, in case two channels have the same name (e.g. from two different platforms). For reference this is what the code looks like now.

https://github.com/Tzahi12345/YoutubeDL-Material/blob/efbf395368c8ffb3af9ac5d432c83eecbe070700/backend/subscriptions.js#L27-L32

I went ahead and updated the latest PR (#873) with the aforementioned fix. Should be merged in the next few days.

Thanks for the bug report!
